backend/queues/makeOfferProcessor.js

```diff
@@ -23,46 +27,55 @@ function attachToQueue() {
  * @param {*} job
  */
 async function processor(job) {
-  const log = (progress, str) => {
-    job.log(str)
+  const queueLog = (progress, str) => {
+    job.queueLog(str)
     job.progress(progress)
   }
 
   const { shopId, encryptedData, paymentCode } = job.data
-  const shop = await getShop(shopId)
-  log(5, 'Load encrypted shop config')
-  const shopConfig = getShopConfig(shop)
-  const network = await getNetwork(shop.networkId)
-  const networkConfig = encConf.getConfig(network.config)
-
-  log(10, 'Creating offer')
-  const lid = ListingID.fromFQLID(shop.listingId)
-  const offer = createOfferJson(lid, encryptedData, paymentCode)
-  const ires = await postOfferIPFS(network, offer)
-
-  log(20, 'Submitting Offer')
-  const web3 = new Web3(network.provider)
-
-  // Use the Shop PK if there is one, otherwise fall back to Network PK.
-  const backendPk = shopConfig.web3Pk || networkConfig.web3Pk
-  const account = web3.eth.accounts.wallet.add(backendPk)
-  const walletAddress = account.address
-  log(22, `using walletAddress ${walletAddress}`)
-  log(25, 'Sending to marketplace')
-  const tx = await offerToMarketplace(
-    web3,
-    lid,
-    network,
-    walletAddress,
-    offer,
-    ires
-  )
-  log(50, JSON.stringify(tx))
+  log.info(`Creating offer for shop ${shopId}`)
+
+  try {
+    const shop = await getShop(shopId)
+    queueLog(5, 'Load encrypted shop config')
+    const network = await getNetwork(shop.networkId)
+    const networkConfig = encConf.getConfig(network.config)
+    const shopConfig = encConf.getConfig(shop.config)
+
+    queueLog(10, 'Creating offer')
+    const lid = ListingID.fromFQLID(shop.listingId)
+    const offer = createOfferJson(lid, encryptedData, paymentCode)
+    const ires = await postOfferIPFS(network, offer)
+
+    queueLog(20, 'Submitting Offer')
+    const web3 = new Web3(network.provider)
+
+    // Use the Shop PK if there is one, otherwise fall back to Network PK.
+    const backendPk = shopConfig.web3Pk || networkConfig.web3Pk
```

Comment:
Maybe we should make this more defensive?  What if a deployment doesn't have a privkey configured for the network yet?

Reply:
good to be defensive. let me check and throw an error...

Reply:
added

Reply:
That doesn't address my concern above?  What happens if neither `shopConfig.web3Pk` nor `networkConfig.web3Pk` are anything on someone's custom deployment?